pandas/tools/plotting.py

```diff
@@ -560,14 +575,14 @@ def f(x):
     for i in range(n):
```

Comment:
the way to get a massive speedup, is do this ALL in a vectorized way, e.g.
pass in the entire numpy array (``df.values``), and do the calculation, then plot in a loop